pkg/image/registry/image/strategy.go

```diff
@@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func (imageStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
I don't think we should allow for updates of `DockerImageReference` containing fully-qualified name, which is part of manifest that can't be changed.